network/p2p/keyTranslator.go

```diff
@@ -92,13 +99,63 @@ func PublicKey(fpk fcrypto.PublicKey) (lcrypto.PubKey, error) {
 		}
 	} else if keyType == lcrypto_pb.KeyType_Secp256k1 {
 		bytes = make([]byte, crypto.PubKeyLenECDSASecp256k1+1) // libp2p requires an extra byte
-		bytes[0] = 4                                           // magic number in libp2p to refer to an uncompressed key
+		bytes[0] = 4                                           // signals uncompressed form as specified in section 4.3.6/7 of ANSI X9.62.
 		copy(bytes[1:], tempBytes)
 	}
 
 	return um(bytes)
 }
 
+// This converts some libp2p PubKeys to a flow PublicKey
+// - the supported key types are ECDSA P-256 and ECDSA Secp256k1 public keys,
+// - libp2p also supports RSA and Ed25519 keys, which Flow doesn't, their conversion will return an error.
+func PublicKeyFromNetwork(lpk lcrypto.PubKey) (fcrypto.PublicKey, error) {
```

Comment:
Could we make sure the namings of these two functions are coherent? They both start with `PublicKey...` and it's a bit confusing.

To me, the clearest naming would be `FlowKeyFromLibP2PKey` and `LibP2PKeyFromFlowKey` or something like that. At the very least, it would be great if the names could be symmetrical.